cirq-superstaq/cirq_superstaq/service.py

```diff
@@ -98,6 +98,54 @@ def counts_to_results(
     return result
 
 
+def _validate_cirq_circuit(circuits: Any) -> None:
+    """Validates that the input is either a single `cirq.Circuit` or a list of `cirq.Circuit`
+    instances.
+
+    Args:
+        circuits: The circuit to run.
+
+    Raises:
+        ValueError: If the input is not a `cirq.Circuit` or a list of
+        `cirq.Circuit` instances.
+    """
+
+    if not (
+        isinstance(circuits, cirq.Circuit)
+        or (
+            isinstance(circuits, Sequence)
+            and all(isinstance(circuit, cirq.Circuit) for circuit in circuits)
+        )
+    ):
+        raise ValueError(
+            "Invalid 'circuits' input. Must be a `cirq.Circuit` or a "
+            "sequence of `cirq.Circuit` instances."
+        )
+
+
+def _validate_get_counts(repetitions: Any) -> None:
+    """Validates that the number of repetitions is positive and
+    an integer.
+
+    Args:
+        repetitions: The number of times to run the circuit.
+
+    Raises:
+        TypeError: If input is not an integer.
+        ValueError: If input is negative.
+    """
+
+    if not (
+        (isinstance(repetitions, int))
+        or (isinstance(repetitions, float) and repetitions.is_integer())
+        or (isinstance(repetitions, str) and repetitions.isdigit())
```

Comment:
isdigit() accepts some unicode values which can't be converted to ints, e.g. `"₁₂₃".isdigit()` returns True but `int("₁₂₃")` fails. isdecimal() is safer:
```suggestion
        or (isinstance(repetitions, (bytes, str)) and repetitions.isdecimal())
```
(also added `bytes` for completeness)